library/sources/http-server/http2/createStreamListener.ts

```diff
@@ -56,17 +56,19 @@ function discoverRouteFromStream(
   if (context && context.route && context.method) {
     const statusCode = parseInt(stream.sentHeaders[":status"] as string);
 
-    if (
-      !isNaN(statusCode) &&
-      shouldDiscoverRoute({
-        statusCode: statusCode,
-        route: context.route,
-        method: context.method,
-      })
-    ) {
-      agent.onRouteExecute(context);
-      // Only count the request if the route is discovered
-      agent.getInspectionStatistics().onRequest();
+    if (!isNaN(statusCode)) {
+      if (
+        shouldDiscoverRoute({
+          statusCode: statusCode,
+          route: context.route,
+          method: context.method,
+        }) ||
+        context.rateLimited // Also discover routes for rate-limited requests
```

Comment:
hmm, when does this happen? I think we don't need it:

should discover route -> the route will show up in the dashboard
user added route manually and then added rate limiting -> if the route is a wildcard, it would not include the rate limited count on the wildcard route? e.g. `ANY /api/*` then the route would be `POST /api/login`, so `POST /api/login` would contain the rate limit counts?